homework/index.js

```diff
@@ -23,24 +10,157 @@
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
+      if (infoRight.firstElementChild === null) {
+        createAndAppend('p', infoRight, {
+          class: 'no-content',
+          text: 'There are no contributions yet.',
+        });
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+    }
+
+    fetchJSON(urlText)
+      .then(contributors => renderContributions(contributors))
+      .catch(error => renderError(error, infoRight));
+  }
+
+  function showRepoInfo(wantedRepo, container) {
+    const infoLeft = createAndAppend('div', container, {
+      class: 'info-left',
+      text: 'Repository information:',
+    });
+    const table = createAndAppend('table', infoLeft);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, `Name`);
+    const secondRow = addRow(tbody, `Description`, wantedRepo.description);
+    addRow(tbody, `Forks`, wantedRepo.forks);
+    addRow(tbody, `Last update`, new Date(wantedRepo.updated_at).toLocaleString());
+    createAndAppend('a', firstRow.lastChild, {
+      text: wantedRepo.name,
+      href: wantedRepo.html_url,
+      target: '_blank',
     });
+    if (secondRow.lastChild.textContent === '') {
+      createAndAppend('span', secondRow.lastChild, {
+        class: 'no-content',
+        text: 'There is NO description.',
+      });
+    }
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  const deleteContents = container => {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  };
+
+  function starter(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('span', header, {
+      class: 'page-title',
+      text: 'HYF Repositories',
+    });
+    const selectEl = createAndAppend('select', header, {
+      class: 'select-list',
+      id: 'repositories_list',
+    });
+    const infoContainer = createAndAppend('div', root, {
+      id: 'info_container',
+      class: 'info-container',
+    });
+
+    function renderResult(repositories) {
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      repositories.forEach((repo, index) => {
+        createAndAppend('option', selectEl, { text: repo.name, value: index });
+      });
+      showRepoInfo(repositories[selectEl.value], infoContainer);
+      showContributions(repositories[selectEl.value].contributors_url, infoContainer);
+      selectEl.addEventListener('change', () => {
+        deleteContents(infoContainer);
+        showRepoInfo(repositories[selectEl.value], infoContainer);
+        setTimeout(
```

Comment:
Why have you used a `setTimeout` here?

Reply:
I forget why! Now I am asking me the same question.
But I noticed that the contributions column has a faster rendering on the screen  with using `setTimeout `. 
I have to think about that again.  It seems that I am going to delete it.